packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -3017,316 +3071,516 @@ void main() {
 
     await tester.pump(); // Wait for autofocus to take effect.
 
-    Future<void> sendKeys(List<LogicalKeyboardKey> keys, {bool shift = false, bool control = false}) async {
-      if (shift) {
-        await tester.sendKeyDownEvent(LogicalKeyboardKey.shiftLeft);
-      }
-      if (control) {
-        await tester.sendKeyDownEvent(LogicalKeyboardKey.controlLeft);
-      }
-      for (LogicalKeyboardKey key in keys) {
-        await tester.sendKeyEvent(key);
-        await tester.pump();
-      }
-      if (control) {
-        await tester.sendKeyUpEvent(LogicalKeyboardKey.controlLeft);
-      }
-      if (shift) {
-        await tester.sendKeyUpEvent(LogicalKeyboardKey.shiftLeft);
-      }
-      if (shift || control) {
-        await tester.pump();
-      }
-    }
-
     // Select a few characters using shift right arrow
     await sendKeys(
+      tester,
       <LogicalKeyboardKey>[
         LogicalKeyboardKey.arrowRight,
         LogicalKeyboardKey.arrowRight,
         LogicalKeyboardKey.arrowRight,
       ],
       shift: true,
+      platform: platform,
     );
 
-    expect(cause, equals(SelectionChangedCause.keyboard));
-    expect(selection, equals(const TextSelection(
-      baseOffset: 0,
-      extentOffset: 3,
-      affinity: TextAffinity.upstream,
-    )));
+    expect(cause, equals(SelectionChangedCause.keyboard), reason: 'on $platform');
```

Comment:
nit: equals doesn't really add anything here, unless you were going to nest matchers

Reply:
Oh, I know that, I just put it in because it reads better.